src/com/google/enterprise/adaptor/fs/WindowsFileDelegate.java

```diff
@@ -406,11 +406,35 @@ private void stopMonitorPaths() {
     }
   }
 
+  /**
+   * An Exponential BackOff that never stops, but does have a maximum sleep
+   * duration between retries.
+   * Modelled after com.google.api.client.util.ExponentialBackOff.
+   */
+  private class NeverEndingExponentialBackOff {
```

Comment:
Why not just ExponentialBackOff? My first thought from the name, before I read the doc comment, was that the exponential interval growth was unbounded, when the opposite is the case. It just doesn't give up. Both the bounded wait and the persistent retries are interesting, but perhaps don't need to be reflected in the name, since it's an isolated, very hard-coded class at this point.